test/controllers/user.controller.test.js

```diff
@@ -1152,183 +1213,199 @@ describe('User Controller', () => {
+                expect(res.body.error).to.be.eql('"User id" is not allowed to be empty');
+                done();
+              });
           });
+        });
       });
     });
+  });
 
-    context('when non admin token is used', () => {
-      it('returns forbidden error', (done) => {
-        request()
-          .put('/api/v1/user/editor/upgrade')
-          .set('authorization', userToken)
-          .send(userInfo)
-          .end((err, res) => {
-            expect(res).to.have.status(403);
-            expect(res.body.success).to.be.eql(false);
-            expect(res.body.message).to.be.eql('You are not permitted to perform this action');
-            done();
-          });
-      });
+  describe('Get all users', () => {
+    const dummyUsers = UserFactory.buildList(16);
+    beforeEach(async () => {
+      adminToken = await addUser(adminUser);
+      userToken = await addUser(regularUser);
+      await User.insertMany(dummyUsers);
     });
 
-    context('when findByIdAndUpdate returns an error', () => {
-      it('returns the error', (done) => {
-        sinon.stub(User, 'findByIdAndUpdate').throws(new Error('Type Error'));
-        request()
-          .put('/api/v1/user/editor/upgrade')
-          .set('authorization', adminToken)
-          .send(userInfo)
-          .end((err, res) => {
-            expect(res).to.have.status(400);
-            expect(res.body.success).to.be.eql(false);
-            done();
-            User.findByIdAndUpdate.restore();
-          });
+    describe('when users exist in db', () => {
+      context('when no parameters are passed', () => {
+        it('returns the default values', (done) => {
+          request()
+            .get('/api/v1/user/all')
+            .set('authorization', adminToken)
+            .end((err, res) => {
+              expectsPaginationToReturnTheRightValues(res, {
+                result: 10,
+                currentPage: 1,
+                limit: 10,
+                offset: 0,
+                total: 18,
+                totalPage: 2,
+              });
+              done();
+            });
+        });
       });
-    });
 
-    context('with invalid data', () => {
-      context('when user id is empty', () => {
-        it('returns an error', (done) => {
+      context('when both page and limit parameters are set', () => {
+        it('returns the given page and limit', (done) => {
           request()
-            .put('/api/v1/user/editor/upgrade')
+            .get('/api/v1/user/all?page=2&limit=5')
             .set('authorization', adminToken)
-            .send({ userId: '' })
             .end((err, res) => {
-              expect(res).to.have.status(412);
-              expect(res.body.success).to.be.eql(false);
-              expect(res.body.message).to.be.eql('Validation Error');
-              expect(res.body.error).to.be.eql('"User id" is not allowed to be empty');
+              expectsPaginationToReturnTheRightValues(res, {
```

Comment:
Using a default value will make this more readable:

```suggestion
               expectsPaginationToReturnTheRightValues(res, { ...expectedPaginationResult, limit:5, page: 2 });
```